src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKey.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotGuardDictionaryRemoveByContainsKey : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1839";
+
+        private static readonly LocalizableString s_localizableTitle = CreateResource(nameof(Resx.DoNotGuardDictionaryRemoveByContainsKeyTitle));
```

Comment:
I haven't made the test but I recall cases where we missed some problems with descriptors because they were wrapped into helper functions like this one instead of being inlined. Tagging @mavasani

Reply:
@Evangelink That could be the `DiagnosticDescriptorCreationAnalyzer`.

Reply:
Do you want me to inline these?

Reply:
@Youssef1313 that's the analyzer that isn't working properly but we had a case where after I inlined the use we had some issues surfacing.

@chucker As I don't recall exactly what were the problems, I'd say it might be safer if you don't mind doing the inline.

Reply:
Sure, no problem.